client.go

```diff
@@ -214,11 +214,10 @@ func (c *client) unassignRoles(ctx context.Context, roleIDs []string) error {
 	var merr *multierror.Error
 
 	for _, id := range roleIDs {
-		var rawResponse *http.Response
-		ctxWithResp := policy.WithCaptureResponse(ctx, &rawResponse)
-		if _, err := c.provider.DeleteRoleAssignmentByID(ctxWithResp, id); err != nil {
+		detailedErr := new(azcore.ResponseError)
+		if _, err := c.provider.DeleteRoleAssignmentByID(ctx, id); err != nil {
 			// If a role was deleted manually then Azure returns a error and status 204
-			if rawResponse != nil && rawResponse.StatusCode == http.StatusNoContent || rawResponse.StatusCode == http.StatusNotFound {
+			if errors.As(err, &detailedErr) && (detailedErr.StatusCode == http.StatusNoContent || detailedErr.StatusCode == http.StatusNotFound) {
```

Comment:
It seems we are only testing the 204 HTTP status. Could we also test the 404 case?

Reply:
Added a test step for the 404 case in [8053ffa](https://github.com/hashicorp/vault-plugin-secrets-azure/pull/213/commits/8053ffa1661044488d419bf15ae1466cc6f8a4fa)